util/reggen/field.py

```diff
@@ -378,29 +355,45 @@ def sw_readable(self) -> bool:
     def sw_writable(self) -> bool:
         return self.swaccess.key != 'ro'
 
-    def apply_alias(self, alias_field: object, where: str) -> None:
-        '''
-        Compare all attributes and replace overridable values.
+    @staticmethod
+    def _gen_attr_err(tst: bool,
+                      attr_name: str,
+                      self_name: str,
+                      alias_name: str,
+                      where: str) -> None:
+        '''Generate attribute mismatch error message'''
+        raise ValueError('Value mismatch for attribute {} between alias field '
+                         '{} and field {} in {}.'.format(attr_name, self_name,
+                                                         alias_name, where))
+
+    def apply_alias(self, alias_field: 'Field', where: str) -> None:
+        '''Compare all attributes and replace overridable values.
 
         This updates the overridable field attributes with the alias values and
         ensures that all non-overridable attributes have identical values.
         '''
-        if not isinstance(alias_field, Field):
-            raise ValueError('alias_field is not of type Field')
-
-        for key, val in self.__dict__.items():
-            # This is an overridable attribute, so replace the value.
-            if key in ALIAS_FIELD_ATTRS:
-                setattr(self, key, getattr(alias_field, key))
-            # This is not an overridable attribute, so ensure values match.
-            elif key in NON_ALIAS_FIELD_ATTRS:
-                if getattr(alias_field, key) != val:
-                    raise ValueError('Value mismatch for attribute {} '
-                                     'between alias field {} and '
-                                     'field {} ({} != {}) in {}.'
-                                     .format(key,
-                                             alias_field.name,
-                                             self.name,
-                                             getattr(alias_field, key),
-                                             val,
-                                             where))
+
+        # Attributes to be crosschecked
+        sname = self.name
+        aname = alias_field.name
+        Field._gen_attr_err(self.bits != alias_field.bits,
```

Comment:
@rswarbrick is this more along the lines of what you had in mind?

Reply:
That looks great! I know it's a bit repetitive, but I think it's probably better because it's now obvious exactly what gets set where.

... although... My strong feelings about `setattr` are because it's such a pain to figure out how a field changed. Maybe `getattr` isn't *so* bad (which would allow you to replace the calls to `_gen_attr_err` with a loop: you could check all fields, providing you did it after overriding the alias-able fields).

I'm not completely sure what's better: maybe worth thinking about? (But I'm also happy with this!)